src/Components/Server/src/ComponentsHub.cs

```diff
@@ -48,15 +50,23 @@ internal CircuitHost CircuitHost
         /// <summary>
         /// Intended for framework use only. Applications should not call this method directly.
         /// </summary>
-        public override async Task OnDisconnectedAsync(Exception exception)
+        public override Task OnDisconnectedAsync(Exception exception)
         {
-            await CircuitHost.DisposeAsync();
+            var circuitHost = CircuitHost;
+            if (circuitHost == null)
+            {
+                return Task.CompletedTask;
+            }
+
+            CircuitHost = null;
+            _circuitRegistry.AddInactiveCircuit(circuitHost);
+            return circuitHost.OnConnectionDownAsync();
         }
 
         /// <summary>
         /// Intended for framework use only. Applications should not call this method directly.
         /// </summary>
-        public async Task StartCircuit(string uriAbsolute, string baseUriAbsolute)
+        public async Task<string> StartCircuit(string uriAbsolute, string baseUriAbsolute)
```

Comment:
Is the string here so that it will get serialized to the client? Think about making this an object instead since we likely will need to add more properties in the future.

Reply:
We *might* not have the ability to send complex types once we switch over to the custom HubProtocol. We could possibly return a JSON payload here instead, but we can always switch that once we need to.

Reply:
Leaving as is for future @pranavkm to figure this out as necessary.